src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -789,7 +796,15 @@ public PercolateShardResponse doPercolate(PercolateShardRequest request, Percola
     private void queryBasedPercolating(Engine.Searcher percolatorSearcher, PercolateContext context, QueryCollector percolateCollector) throws IOException {
         Filter percolatorTypeFilter = context.indexService().mapperService().documentMapper(TYPE_NAME).typeFilter();
         percolatorTypeFilter = context.indexService().cache().filter().cache(percolatorTypeFilter);
-        XFilteredQuery query = new XFilteredQuery(context.percolateQuery(), percolatorTypeFilter);
+
+        final Filter filter;
+        if (context.aliasFilter() != null) {
+            filter = new AndFilter(Arrays.asList(context.aliasFilter(), percolatorTypeFilter));
```

Comment:
yes in the and filter that does matter. My reasoning was that the alias filter is more specific and will exclude more docs and that is why I put it first... Maybe I should use the bool filter instead, where such decisions are made automatically. (in master we can use bool query :) which is even better)